teps/0109-better-structured-provenance-retrieval-in-tekton-chains.md

```diff
@@ -0,0 +1,565 @@
+
+Ultimately, we are looking for support for as many format types of inputs and outputs as possible, but in this doc, we are going to focus on the a list that’s broadly needed.
+
+For `inputs`, if we look at the requirements of SLSA in terms of sources, only SLSA L4 requires [Dependencies complete](https://slsa.dev/spec/v0.1/requirements#dependencies-complete). At the current point, if we only worry about the requirements lower than SLSA L4, support that covers VCS and OCI images is a good start. In in-toto provenance, all the inputs should be stored in the [materials](https://github.com/in-toto/attestation/tree/main/spec#predicate-conventions) under in-toto provenance. The required format is:
+```
+{
+        "uri": "<URI>",
+        "digest": { /* DigestSet */ }
+ }
+```
+The other specific requirements of these fields can be found in [in-toto attestation spec](https://github.com/in-toto/attestation/tree/main/spec). Another thing to be aware is that For `digest`, it has a key for the digest type and has a value for that key. Git, Mercurial, Bazaar and Subversion have full VCS location support. CVS, Perforce are providing this VCS location as plain URLs. For VCS, the list of VCS that the community are looking into is: Git, Mercurial, Subversion, Bazaar, CVS, Perforce. Currently Chains only supports Git revisions for VCS and OCI images as signable targets for inputs.
+
+For the `outputs`, we are looking at the following support, as these are the artifact types supported in Artifact Registries. However, the list should definitely be expanded as this doc is spreading the discussions in the communities to see which should be prioritized. Currently, we have the following under considerations: Python, Maven, Go, NodeJS and OCI images. Note that OCI images are the ones we currently support.
+
+#### Concrete Use Cases
+All below examples will be generating in-toto attestation.
+ 1. Use Git commits as sources, and generates in-toto provenance for a TaskRun that builds an image.
+   
+    A Task example `without` structured results and params support can be like:
+    ``` yaml
+    apiVersion: tekton.dev/v1beta1
+    kind: Task
+    spec:
+      params:
+        - name: CHAINS-GIT_COMMIT
+          type: string
+          description: git commit sha
+        - name: CHAINS-GIT_URL
+          type: string
+          description: git commit url
+        - name: FOO_IMAGE
+          type: string
+        - description: artifact image
+      ...
+      results:
+        - name: IMAGE_DIGEST
+          description: digest of image
+        - name: IMAGE_URL
+          description: url of image
+    ```
+    A Task example `with` structured results support will be like:
+    ``` yaml
+    apiVersion: tekton.dev/v1beta1
+    kind: Task
+    spec:
+      ...
+      results:
+        - name: SIGNABLE-VCS_$(NAME)
```

Comment:
is `$(NAME)` actual interpolation, or just a placeholder?

Reply:
Ah this is just a placeholder in the doc.